canvas_modules/harness/cypress/support/properties/properties-cmds.js

```diff
@@ -217,3 +218,165 @@ Cypress.Commands.add("clickButtonInTable", (buttonName, tableControlId) => {
 			}
 		});
 });
+
+// Expression control commands
+Cypress.Commands.add("getAutoCompleteCountForText", (text) => {
+	// Select all and delete existing text in expression editor
+	cy.useCtrlOrCmdKey()
+		.then((selectedKey) => {
+			cy.get(".properties-expression-editor")
+				.find(".CodeMirror")
+				.find(".CodeMirror-code")
+				.type(selectedKey + "{a}{del}");
+
+			// Type text and ctrl + space to display hints
+			cy.get(".CodeMirror-code")
+				.type(text + "{ctrl} ");
+
+			cy.get(".CodeMirror-hints")
+				.eq(0)
+				.find("li")
+				.its("length");
+		});
+});
+
+Cypress.Commands.add("selectFirstAutoCompleteForText", (text) => {
+	// Select all and delete existing text in expression editor
+	cy.useCtrlOrCmdKey()
+		.then((selectedKey) => {
+			cy.get(".properties-expression-editor")
+				.find(".CodeMirror")
+				.find(".CodeMirror-code")
+				.type(selectedKey + "{a}{del}");
+
+			// Type text and ctrl + space to display hints
+			cy.get(".CodeMirror-code")
+				.type(text + "{ctrl} ");
+
+			// select the first one in the list of hints and make sure it is the text
+			cy.get(".CodeMirror-hints")
+				.eq(0)
+				.find("li")
+				.eq(0)
+				.click();
+		});
+});
+
+Cypress.Commands.add("enterTextInExpressionEditor", (text) => {
+	// Select all and delete existing text in expression editor
+	cy.useCtrlOrCmdKey()
+		.then((selectedKey) => {
+			cy.get(".properties-expression-editor")
+				.find(".CodeMirror")
+				.find(".CodeMirror-code")
+				.type(selectedKey + "{a}{del}");
+
+			// Type text
+			cy.get(".CodeMirror-code")
+				.type(text);
+		});
+});
+
+Cypress.Commands.add("clickValidateLink", () => {
+	cy.get(".properties-expression-validate")
+		.find(".validateLink")
+		.click();
+});
+
+Cypress.Commands.add("clickValidateLinkInSubPanel", (panelName) => {
+	cy.getWideFlyoutPanel(panelName)
+		.then((wideFlyoutPanel) => {
+			cy.wrap(wideFlyoutPanel)
+				.find(".properties-expression-validate")
+				.find(".validateLink")
+				.click();
+		});
+});
+
+Cypress.Commands.add("clickExpressionBuildButtonForProperty", (propertyName) => {
+	cy.get(`div[data-id='properties-ci-${propertyName}']`)
+		.find(".properties-expression-button")
+		.click();
+});
+
+Cypress.Commands.add("selectFieldFromPropertyInSubPanel", (fieldName, propertyName, panelName) => {
+	cy.getWideFlyoutPanel(panelName)
```

Comment:
Same as above.  I would think opening the wideFlyoutPanel should be in the test.

Reply:
I think this also needs to use the propertyName.  I don't think it's being used right now in the selector.